src/module-kount-control/Model/ControlApi/TrustedDevice.php

```diff
@@ -0,0 +1,169 @@
+<?php
+
+/**
+ * Created by Swarming Technology, LLC.
+ * Project: Kount
+ */
+
+namespace Swarming\KountControl\Model\ControlApi;
+
+use Swarming\KountControl\Api\ServiceInterface;
+
+class TrustedDevice extends AbstractService implements ServiceInterface
+{
+    const ENDPOINT_URI = '/trusted-device';
+    const TRUSTED_DEVICE = "TRUSTED";
+
+    /**
+     * @var string
+     */
+    private $deviceId;
+
+    /**
+     * @inheritdoc
+     */
+    public function preparePayload($sessionId, $clientId)
+    {
+        $userId = $this->customerSession->getCustomerId();
+        return [
+            'body' => json_encode(
+                [
+                    'clientId' => $clientId,
+                    'sessionId' => $sessionId,
+                    'userId' => $userId,
+                    'trustState' => self::TRUSTED_DEVICE
+                ]
+            )
+        ];
+    }
+
+    /**
+     * @return string
+     */
+    public function getDeviceId()
+    {
+        return $this->deviceId;
+    }
+
+    /**
+     * @param $deviceId
+     */
+    public function setDeviceId($deviceId)
+    {
+        $this->deviceId = $deviceId;
+    }
+
+    /**
+     * @inheritdoc
+     */
+    public function postData(array $payload)
+    {
+        $response = null;
+        $deviceId = $this->getDeviceId();
+        $apiKey = $this->getApiKey();
+        $payload = $this->addAuthToPayload($payload);
+        $payloadArray = json_decode($payload['body'], true);
+        if (
+            isset($payloadArray['clientId'])
+            && isset($payloadArray['sessionId'])
+            && isset($payloadArray['userId'])
+        ) {
+            $clientId = $payloadArray['clientId'];
+            $sessionId = $payloadArray['sessionId'];
+            $userId = $payloadArray['userId'];
+        } else {
+            throw new \Swarming\KountControl\Exception\ParamsException(__(
+                'KountControl: lost GET params. $customerId, $sessionId or $userId not set'
+            ));
+        }
+
+        // Check whether exists trusted Device record with this device ID for merchant with this client ID
+        $responseBody = '';
+        try {
+            $response = $this->client->request(
+                'get',
+                'https://api-sandbox.kount.com/trusted-device/sessions/' . $sessionId
+                    . '/users/' . $userId . '/clients/' . $clientId,
+                ['headers' => ['Authorization' => 'Bearer ' . $apiKey]]
+            );
+            $responseBody = $response->getBody()->getContents();
+
+            $this->logger->info(__('KountControl: GET response from ' . $this->getUri() . ' got '
+                . $response->getStatusCode() . ' status code and body: ' . $responseBody));
+        } catch (\GuzzleHttp\Exception\GuzzleException $e) {
+            if ($e->getResponse()->getStatusCode() && $e->getMessage()) {
+                $this->logger->error(__('KountControl: Response from ' . $this->getUri() . ' got '
```

Comment:
I think this block of code is seen in every catch in this function, can you move this to a separate method to minimize redundant code.  For example in the catch block call $this->handleException($e);

Reply:
fixed